### ISSUE TITLE
Remove shipping cost from cart and checkout before carrier selection

### ISSUE THREAD

**Issue by mrports** [PrestaShop/PrestaShop#29046]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

having an overview of how much you will be paying is "nice". but "not necessary" and not worth the troubles associated
Having the shipping cost calculated on a loose parameters like default carrier and default country is problematic when neither these parameters are appropriate. there are many cases when the price can be misleading and worse when the price is wrongly calculated that lead the customer to abandon the website before reaching the checkout .
just check the forum to see how many discussions are created to fix this problem

### Describe the solution you'd like

just make an option to the admin to **display** or **hide** the shipping cost line from cart and checkout before carrier selection.

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by Hlavtox** [comment]
Yes, this is requested lot by my clients, sometimes it doesn't show things correctly.

- `Cheapest carrier` is unusable, because it will always select personal pickup, that is free.
- `Default carrier XXX` is unusable, because the carrier options differ for different products.

Ended up using modifying `Cart::getDeliveryOption` function to select best price excluding personal pickup.

Option to hide the shipping would be also good, but it would need to modify the totals in cart.

**Comment by hibatallahAouadni** [comment]
Hello @mrports 

Thank you for your suggestion. The Maintainer Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

PS: thanks @Hlavtox for your quick reply :superhero: and we can't wait for your PR :rocket: 

Thank you
